pypesto/petab/importer.py

```diff
@@ -415,6 +436,37 @@ def create_objective(
             par_ids = [x for x in par_ids if x not in inner_parameter_ids]
 
             # FIXME: currently not supported with hierarchical
+            if 'guess_steadystate' in kwargs and kwargs['guess_steadystate']:
+                warnings.warn(
+                    "`guess_steadystate` not supported with hierarchical optimization. Disabling `guess_steadystate`."
+                )
+            kwargs['guess_steadystate'] = False
+
+        if self._ordinal:
```

Comment:
shouldn't be there be separate handling of `self._ordinal={True}`x`self._hierarchical={True,False}` conditions?

Reply:
As it is currently implemented, ordinal measurements can be dealt with only hierarchically, with optimal scaling parameters (category parameters) being optimized in the inner loop. 

It would be possible to implement the possibility of optimizing them non-hierarchically, but it would take time, and as it would probably be much slower because sometimes there can be as many optimal scaling parameters as many measurements there are, I would say it's probably not worth it.

But I agree that the checks here are ambiguous (or naming). Because it would then make sense to have `self._hierarchical=True` obligatory for `self._ordinal=True`. But since the optimal scaling method of dealing with ordinal data is so different from the hierarchical manner of dealing with scaling, offset, and noise parameters, it made the most sense to separate it completely and with that to have different checks. 

I guess what would make the most sense would be to have:
`if self._hierarchical and not self._ordinal` -> Go into the hierarchical approach of optimizing scaling, offset, and noise parameters
`if self._hierarchical and self._ordinal` -> Go into Optimal Scaling for dealing with ordinal data.

But then again, don't know if this is just overcomplicating over a simple issue, and will be even worse with the later addition of `self._nonlinear_monotone`.

Reply:
Or it's better even:
`
if self._hierarchical and not self._ordinal
   ... optimal scaling & ordinal
elif self._hierarchical
` ... scaling offset & noise

Reply:
I think if there is no intention to implement a non-hierarchical ordinal approach it's fine to set `self._hierarchical=True` in case `ordinal==True`, but this should be noted in the docstrings for both `ordinal` and `hierarchical`.